src/linter/rules/unsafe_undefined.zig

```diff
@@ -153,6 +153,10 @@ fn undefinedDefault(ctx: *LinterContext, undefined_tok: TokenIndex) Error {
     e.help = Cow.static("If this really can be `undefined`, do so explicitly during struct initialization.");
     return e;
 }
+fn getTokenSource(ctx: *LinterContext, token_idx: TokenIndex) []const u8 {
+    const span = ctx.ast().tokenToSpan(token_idx);
```

Comment:
use `ctx.semantic.tokenSlice`. We should always avoid `ast.tokenToSpan`; see the comment around `tokenSlice` for rationale